config/routes.rb

```diff
@@ -29,11 +29,22 @@
   patch "/veterinarians/:id", to: "veterinarians#update"
   delete "/veterinarians/:id", to: "veterinarians#destroy"
 
-  get "/shelters/:shelter_id/pets", to: "shelters#pets"
+  get "/shelters/:shelter_id/pets", to: "shelter_pets#index"
   get "/shelters/:shelter_id/pets/new", to: "pets#new"
   post "/shelters/:shelter_id/pets", to: "pets#create"
 
-  get "/veterinary_offices/:veterinary_office_id/veterinarians", to: "veterinary_offices#veterinarians"
+  get "/veterinary_offices/:veterinary_office_id/veterinarians", to: "veterinary_office_veterinarians#index"
   get "/veterinary_offices/:veterinary_office_id/veterinarians/new", to: "veterinarians#new"
   post "/veterinary_offices/:veterinary_office_id/veterinarians", to: "veterinarians#create"
+
+  get "/applications/new", to: "applications#new"
+  get "/applications/:id", to: "applications#show"
```

Comment:
See note in applications_controller about this routing. For adding a pet to an app, you could route the button to:
`post "/applications/:id/pets", to: "pet_applications#create"`

Similarly, when you approve an application, you could route that approval button to
`patch "/applications/:app_id/pets/:pet_id", to: "pet_applications#update"`